zkevm-circuits/src/evm_circuit/util/common_gadget/tx_l1_fee.rs

```diff
@@ -158,8 +289,32 @@ impl<F: Field> TxL1FeeGadget<F> {
         ]
         .map(|word| from_bytes::expr(&word.cells[..N_BYTES_U64]));
 
+        #[cfg(feature = "l1_fee_curie")]
+        let [l1_blob_basefee, commit_scalar, blob_scalar] = [
+            &l1_blob_basefee_word,
+            &commit_scalar_word,
+            &blob_scalar_word,
+        ]
+        .map(|word| from_bytes::expr(&word.cells[..N_BYTES_U64]));
+
         // <https://github.com/scroll-tech/go-ethereum/blob/49192260a177f1b63fc5ea3b872fb904f396260c/rollup/fees/rollup_fee.go#L118>
-        let tx_l1_gas = tx_data_gas_cost + TX_L1_COMMIT_EXTRA_COST.expr() + fee_overhead;
+        let tx_l1_gas = if cfg!(feature = "l1_fee_curie") {
+            0.expr()
+        } else {
+            tx_data_gas_cost + TX_L1_COMMIT_EXTRA_COST.expr() + fee_overhead
+        };
+
+        // TODO: new formula for curie
+        #[cfg(feature = "l1_fee_curie")]
+            cb.require_equal(
+            "commitScalar * l1BaseFee + blobScalar * _data.length * l1BlobBaseFee == tx_l1_fee * 10e9 + remainder",
+            commit_scalar * base_fee + blob_scalar * tx_signed_length * l1_blob_basefee,
```

Comment:
Is this sound? Anywhere we range check remainder?

Reply:
there is no check remainder for old l1 fee such field,  should be a bug, will file an issue to trace this and fix later.

Reply:
@noel2004 can have a look

Reply:
filed bug https://github.com/scroll-tech/zkevm-circuits/issues/1307

Reply:
Have purposed a fixing in https://github.com/scroll-tech/zkevm-circuits/pull/1308/files

Reply:
merged into this pr .